### PR TITLE
Update notch with pre/post batch

### PR DESCRIPTION
**Pull request by Hwurzburg**

### CODE REVIEW

File: common/source/docs/common-imu-notch-filtering.rst [hunk 1]
```diff
@@ -51,9 +61,11 @@ Double/Triple-Notch
...
+.. note:: double notch option is no longer recommended since the triple notch option has been added.With a double notch, the maximum attenuation is either side of the center frequency, so on smaller aircraft with a very pronounced peak their use is usually counter productive.
+
+.. note:: Each notch has some CPU cost so if you configure multiple notches you can end up with many notches on your aircraft.For example, triple notches, using ESC telemetry will result in 3 notches per motor or 12 total notches. For example, with F4 cpus this should be acceptable, but enabling a second group of triple notches with :ref:`INS_HNTC2_ENABLE<INS_HNTC2_ENABLE>` could cause problems.
```
Comment (andyp1per) [inline common/source/docs/common-imu-notch-filtering.rst:66]: one of the changes in 4.3 is the ability to have harmonics on notch-per-motor - you might want to mention this somewhere

**Review by andyp1per — CHANGES_REQUESTED** [review]
Looks good. You might want to get Leonard to comment on the phase lag stuff as that is not the total story.

File: common/source/docs/common-imu-batchsampling.rst [hunk 2]
```diff
@@ -15,14 +15,14 @@ Samples are typically taken at the same rate as gyro updates provided to the aut
...
+The backend rate has been made configurable for IMUs that support fast sampling (i.e. Invensense sensors) so that higher rates can be supported. The backend rate can be configured by setting :ref:`INS_GYRO_RATE<INS_GYRO_RATE>`. The default 0 gives the same behaviour as previous firmware versions. A rate of N gives a backend rate of 2^N Khz, so for instance a value of 1 gives a backend rate of 2Khz ,etc.
 
 .. note:: This raises the rate at which all gyro filters are run and can be computationally expensive depending on the number of notches configured. Values above 0 are only recommend on F7 or H7 platforms and values above 1 should only be used with careful tuning. This is automatically done for those platforms. This allows capturing frequencies at 1KHz and below on these autopilots.
 
 Pre-Flight Setup
 ================
 
-- Set :ref:`INS_LOG_BAT_OPT<INS_LOG_BAT_OPT>` = 0 to do pre-filter 1KHz sampling
+- Set :ref:`INS_LOG_BAT_OPT<INS_LOG_BAT_OPT>` = 4 to do pre and post-filter 1KHz sampling
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-imu-batchsampling.rst:25]: is this necessary? "as the logging backend is fast enough" worries me...don't want any failures for users...didn't we up the overall sampling to 1minute fro 30sec ? doesn't this provide enough samples??

File: common/source/docs/common-imu-batchsampling.rst [hunk 3]
```diff
@@ -31,27 +31,36 @@ Pre-Flight Setup
 Flight and Post-Flight Analysis
 ===============================
 
-- Perform a regular flight (not just a gentle hover) of at least 30 seconds and :ref:`download the dataflash logs <common-downloading-and-analyzing-data-logs-in-mission-planner>`
-- Open Mission Planner, press Ctrl-F, press the FFT button, press "new DF log" and select the .bin log file downloaded above
+- Perform a regular flight (not just a gentle hover) of at least 1 minute and :ref:`download the dataflash logs <common-downloading-and-analyzing-data-logs-in-mission-planner>`
+- Open Mission Planner, under SETUP/ADVANCED, press the FFT button, press "IMU Barch Sample" and select the .bin log file downloaded above
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-imu-batchsampling.rst:35]: thanks..done

File: common/source/docs/common-imu-batchsampling.rst [hunk 4]
```diff
@@ -31,27 +31,36 @@ Pre-Flight Setup
...
 
-.. image:: ../../../images/pre-tune-fft.png
-    :target:  ../_images/pre-tune-fft.png
+.. image:: ../../../images/fft-pre.jpg
+    :target:  ../_images/fft-pre.jpg
     :width: 450px
 
-- Accelerometer data appears in the top left window with the vertical axis showing the amplitude and horizontal axis showing the frequency.  The amplitude is not scaled to a useful value, meaning the graph is useful for determining the frequency of the vibration, but not whether the levels are too high or not.  Vibration at frequencies above 300Hz may lead to attitude or position control problems.
+- Four graphs are shown: ACC0- Accelerometer spectrum before filtering, ACC1- Accelerometer spectrum before filtering, GYR0 -Gyro spectrum before filtering, and GYR1- Gyro spectrum after filtering.  Vibration at frequencies above 100Hz may lead to attitude or position control problems.
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-imu-batchsampling.rst:47]: cut an paste error..thanks

File: common/source/docs/common-imu-batchsampling.rst [hunk 5]
```diff
@@ -31,27 +31,36 @@ Pre-Flight Setup
...
+- A pronounced noise peak at 180Hz is evident from the motor's in hover, with a smaller second harmonic showing.
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-imu-batchsampling.rst:48]: fixed

File: common/source/docs/common-imu-batchsampling.rst [hunk 6]
```diff
@@ -31,27 +31,36 @@ Pre-Flight Setup
...
+- Notice that the noise has been dramatically reduced by the lowpass filters on the sensors, but at the cost of significant phase lag that will have reduced how tight the tune was made. It should be possible to "notch" out that 180Hz motor noise spike and increase the gyro low pass filter to 60 or 80Hz to allow a tighter tune to be done. The accelerometer path is not as critical since it forms an outer loop around the gryo based rate PID controllers. Leaving its lowpass filter at 10Hz does not affect the ability of the vehicle to reject disturbances quickly, which is the rate controller's main task.
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-imu-batchsampling.rst:49]: fixed

File: common/source/docs/common-imu-notch-filtering.rst [hunk 7]
```diff
@@ -10,11 +10,21 @@ As discussed under the :ref:`Vibration Damping<common-vibration-damping>` topic,
 
 To the autopilot, vibration noise looks like any other disturbance (e.g. wind, turbulence, control link slop, etc.) that the autopilot must compensate for in order to control the aircraft. This prevents optimum tuning of the attitude control loops and decreased performance.
 
+ArduPilot provides two filtering mechanisms for noise. Lowpass filters on the accelerometer signals, controlled by the :ref:`INS_ACCEL_FILTER<INS_ACCEL_FILTER>`, and  the gyro signals, controlled by :ref:`INS_GYRO_FILTER<INS_GYRO_FILTER>`, and Harmonic Notch Filters on the gyro signals.
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-imu-notch-filtering.rst:13]: I think that is TMI...it does not affect how the user adjusts anything AFAIK...the EKF is naturally a huge noise filterer by its nature..

**Comment by Hwurzburg** [comment]
> Looks good. You might want to get Leonard to comment on the phase lag stuff as that is not the total story.
@lthall Leonard, anything additional needed? should be directly related to how a user would setup the notches...the phase lag comment is to motivate a user to increase the lowpasses and use notches, if possible, rather than leave them as low frequency...

File: common/source/docs/common-imu-notch-filtering.rst (hunk 1 above)
Comment (Hwurzburg) [reply common/source/docs/common-imu-notch-filtering.rst]: see line 44, is this what you had in mind?

File: common/source/docs/common-imu-batchsampling.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (andyp1per) [reply common/source/docs/common-imu-batchsampling.rst:25]: it's not the same - LGIN is effectively the gap between batches, the bigger the gap the fewer batches but also the more likely that the averaging won't make sense. but to be clear, I am saying you should mention it - not that we necessarily need to ask users to do this out of the box

File: common/source/docs/common-imu-notch-filtering.rst (hunk 1 above)
Comment (andyp1per) [reply common/source/docs/common-imu-notch-filtering.rst]: no, not the same. now if you set HMNCS you will get that number of harmonics on *each* motor. It use to be that you only got a single harmonic. So its possible to get a very large number of notches this way. For instance quad with triple notch and 3 harmonics == 4 x 3 x 3 == 36 notches

File: common/source/docs/common-imu-batchsampling.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-imu-batchsampling.rst:25]: added verbage at line 22

File: common/source/docs/common-imu-notch-filtering.rst (hunk 1 above)
Comment (Hwurzburg) [reply common/source/docs/common-imu-notch-filtering.rst]: added notes in mode and triple notch section

File: common/source/docs/common-imu-notch-filtering.rst [hunk 8]
```diff
@@ -31,15 +41,17 @@ Key to the dynamic notch filter operation is control of its center frequency. Th
...
 
 All of the above are repeated, independently, for the second notch and are prefaced with ``INS_HNTC2_`` instead of ``INS_HNTCH_``. The following will explain setup for the first set of notches.
 
 Number of Harmonics Filtered
 ============================
 
-- Set :ref:`INS_HNTCH_HMNCS <INS_HNTCH_HMNCS>` to enable up to three harmonics (multiples of the center frequency) for notches.
+- Set :ref:`INS_HNTCH_HMNCS <INS_HNTCH_HMNCS>` to enable up to three harmonic (multiples of the center frequency) notches. For ESC :ref:`INS_HNTCH_MODE <INS_HNTCH_MODE>` tracking, each motor will get a set of these notches. If an octocopter sets up all three harmonics, this results in 8 x 3 = 24 notch filters. Enabling triple notches (see below) would result in 72 filters! This would most certainly cause excessive cpu loading and performance issues.
```
Comment (andyp1per) [inline common/source/docs/common-imu-notch-filtering.rst:52]: True for FFT as well.

**Review by andyp1per — APPROVED** [review]
Just a comment about FFT, but looks good

File: common/source/docs/common-imu-notch-filtering.rst (hunk 8 above)
Comment (Hwurzburg) [reply common/source/docs/common-imu-notch-filtering.rst]: updated

**Comment by Hwurzburg** [comment]
merging

**Pull request merged**